### PR TITLE
CSS: Mask-based loss and other minor modifications.

### PR DESCRIPTION
**Pull request by ig177**
Additional modifications:
* Introduced a noise weighting hyperparameter.
* Added side info outputs to the loss function and separate_and_stitch() to gain better insight into training and inference.
* Refactored css.py a little.

### CODE REVIEW

**Review by nidleo — APPROVED** [review]
It's best to also address the mask-based loss in description and config.
